### ISSUE TITLE
Testing TrustSVD with a separate TestSet

### ISSUE THREAD

**Issue by Sam131112** [guoguibing/librec#21]
When I run TrustSVD with following conf setup 

dataset.ratings.wins=.\demo\Datasets\FilmTrust\ratings.txt
dataset.ratings.lins=/Users/sam/Downloads/librec/librec/demo/Datasets/FilmTrust/pets_pets_train.txt

dataset.social.wins=.\demo\Datasets\FilmTrust\trust.txt
dataset.social.lins=/Users/sam/Downloads/librec/librec/demo/Datasets/FilmTrust/pets_pets_user_user_con1.txt

ratings.setup=-columns 0 1 2 -threshold -1

recommender=TrustSVD
evaluation.setup=cv -k 5 -p on --rand-seed 1 --test-view all
item.ranking=off -topN -1 -ignore -1

num.factors=10
num.max.iter=100

learn.rate=0.001 -max -1 -bold-driver
reg.lambda=0.1 -u 1.2 -i 1.2 -b 1.2 -s 0.9

output.setup=on -dir ./demo/Results/

it runs fine 
However when i replace evaluation setup by 
evaluation.setup = test-set -f /Users/sam/Downloads/librec/librec/demo/Datasets/FilmTrust/pets_pets_test.txt 

it gives following error 

java.lang.ArrayIndexOutOfBoundsException: 1137
    at librec.data.SparseMatrix.columnSize(SparseMatrix.java:563)
    at librec.rating.TrustSVD.initModel(TrustSVD.java:83)
    at librec.intf.Recommender.execute(Recommender.java:318)
    at librec.main.LibRec.runAlgorithm(LibRec.java:485)
    at librec.main.LibRec.run(LibRec.java:399)
    at librec.main.LibRec.execute(LibRec.java:148)
    at librec.main.Demo.execute(Demo.java:197)
    at librec.main.Demo.main(Demo.java:40)

I cant figure out what went wrong here

**Comment by guoguibing** [comment]
I guess this is a bug of LibRec. When training and test files are separated, some users existing only in test set may not exist in the training set.
